content/en/blog/2023/contributing-to-otel-demo/index.md

```diff
@@ -0,0 +1,1026 @@
+- Load Generator
+  UI: [http://localhost:8080/loadgen/](http://localhost:8080/loadgen/)
+- Jaeger
+  UI: [http://localhost:8080/jaeger/ui/](http://localhost:8080/jaeger/ui/)
+- Tracetest UI: [http://localhost:11633/](http://localhost:11633/), only when
+  using `make start-odd`
+
+To run a test against the Payment Service, I’ll use a YAML file and trigger it
+with the
+[Tracetest CLI](https://docs.tracetest.io/getting-started/installation#install-the-tracetest-cli).
+Alternatively, you can also build tests visually in the Tracetest UI
+on [http://localhost:11633/](http://localhost:11633/).
+
+> _Here’s a
+> [short guide on how to create tests programatically](https://docs.tracetest.io/getting-started/open#creating-trace-based-tests-programatically)
+> with Tracetest._
+
+Next, let’s move on to explaining and adding code instrumentation in the
+`paymentservice`.
+
+### Adding OpenTelemetry to the Payment Service
+
+I’ve prepared a fork with detailed code examples and three demos. This will help
+you understand how to add OpenTelemetry code instrumentation.
+
+```bash
+git clone https://github.com/kubeshop/opentelemetry-demo.git
+cd opentelemetry-demo/
+make start-odd
+```
+
+Here’s a snippet of what configuring
+[automatic instrumentation for Node.js looks like in the `src/paymentservice/opentelemetry.js` file](https://github.com/kubeshop/opentelemetry-demo/blob/main/src/paymentservice/opentelemetry.js).
+
+```jsx
+// src/paymentservice/opentelemetry.js
+
+const opentelemetry = require('@opentelemetry/sdk-node');
+const {
+  getNodeAutoInstrumentations,
+} = require('@opentelemetry/auto-instrumentations-node');
+const {
+  OTLPTraceExporter,
+} = require('@opentelemetry/exporter-trace-otlp-grpc');
+const {
+  OTLPMetricExporter,
+} = require('@opentelemetry/exporter-metrics-otlp-grpc');
+const { PeriodicExportingMetricReader } = require('@opentelemetry/sdk-metrics');
+const {
+  alibabaCloudEcsDetector,
+} = require('@opentelemetry/resource-detector-alibaba-cloud');
+const {
+  awsEc2Detector,
+  awsEksDetector,
+} = require('@opentelemetry/resource-detector-aws');
+const {
+  containerDetector,
+} = require('@opentelemetry/resource-detector-container');
+const { gcpDetector } = require('@opentelemetry/resource-detector-gcp');
+const {
+  envDetector,
+  hostDetector,
+  osDetector,
+  processDetector,
+} = require('@opentelemetry/resources');
+
+const sdk = new opentelemetry.NodeSDK({
+  // OTLPTraceExporter() uses the env var "OTEL_EXPORTER_OTLP_ENDPOINT" when not explicitly set.
+  traceExporter: new OTLPTraceExporter(),
+  instrumentations: [getNodeAutoInstrumentations()],
+  metricReader: new PeriodicExportingMetricReader({
+    exporter: new OTLPMetricExporter(),
+  }),
+  resourceDetectors: [
+    containerDetector,
+    envDetector,
+    hostDetector,
+    osDetector,
+    processDetector,
+    alibabaCloudEcsDetector,
+    awsEksDetector,
+    awsEc2Detector,
+    gcpDetector,
+  ],
+});
+sdk.start();
+```
```

Comment:
Can you make this (and other) code examples a little bit compacter? I think you can write more things in one line than you do it in the original code